src/Dumplie/Application/Query/Customer/Result/Cart.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+declare (strict_types = 1);
+
+namespace Dumplie\Application\Query\Customer\Result;
+
+use Dumplie\Application\Exception\InvalidArgumentException;
+
+final class Cart
+{
+    /**
+     * @var CartItem[]|array
+     */
+    private $items;
+
+    /**
+     * @var float
+     */
+    private $totalPrice;
+
+    /**
+     * @param array|CartItem[] $items
+     * @throws InvalidArgumentException
+     */
+    public function __construct(array $items = [])
+    {
+        $this->totalPrice = 0;
+
+        foreach ($items as $item) {
+            if (!$item instanceof CartItem) {
+                throw new InvalidArgumentException();
+            }
+
+            $this->totalPrice += $item->price();
+        }
+
+        $this->items = $items;
+    }
+
+    /**
+     * @return bool
+     */
+    public function isEmpty() : bool
+    {
+        return !count($this->items);
+    }
+
+    /**
+     * @return float
+     */
+    public function totalPrice() : float
+    {
+        return $this->totalPrice;
+    }
+
+    /**
+     * @return string
+     */
+    public function currency() : string
+    {
+        return $this->isEmpty() ? null : $this->items[0]->currency();
```

Comment:
I would avoid mixing currencies and doing conversions between them.
This may lead to problems with displaying prices based on current exchange rate from NBP or other sources - in the end, such rates will be different and depend on customers bank rates.

Also, prices in different currencies may varry not only because of exchange rate also because of different tax rules or local law.

Therefore all prices should be independent.